module/Education/view/education/education/course.phtml

```diff
@@ -6,86 +6,96 @@ $this->headTitle($this->translate('Education'));
-                                case Exam::EXAM_TYPE_FINAL:
-                                    $name = $this->translate('Examination from %s (%s)');
-                                    break;
-                                case Exam::EXAM_TYPE_INTERMEDIATE_TEST:
-                                    $name = $this->translate('Intermediate test from %s (%s)');
-                                    break;
-                                case Exam::EXAM_TYPE_ANSWERS:
-                                    $name = $this->translate('Answers from %s (%s)');
-                                    break;
-                                case Exam::EXAM_TYPE_OTHER:
-                                    $name = $this->translate('Other exam material from %s (%s)');
-                                    break;
-                            }
-                            ?>
-                            <?= sprintf($name, $exam->getDate()->format('Y-m-d'), $exam->getLanguage()) ?>
-                        <?php endif ?>
-                    </a></li>
-            <?php endforeach; ?>
-        </ul>
+        <div class="row">
+            <div class="col-md-12">
+                <h1><?= $this->escapeHtml($course->getCode()) ?> - <?= $this->escapeHtml($course->getName()) ?></h1>
+            </div>
+            <?php
+            if (null !== $course->getUrl()):
+                ?>
+                <div class="col-md-12">
+                    <a href="<?= $course->getUrl() ?>"><?= $this->escapeHtml($course->getUrl()) ?></a>
+                </div>
+            <?php
+            endif;
+            ?>
+        </div>
+        <hr>
+        <div class="row">
+            <div class="col-md-4">
+                <h2><?= $this->translate('Subcourses') ?></h2>
+                <?php if (!$course->getChildren()->isEmpty()): ?>
+                    <ul>
+                        <?php foreach ($course->getChildren() as $child): ?>
+                            <li>
+                                <a href="<?= $this->url('education/course', ['code' => $child->getCode()])?>">
+                                    <strong><?= $this->escapeHtml($child->getCode()) ?></strong> <?= $this->escapeHtml($child->getName()) ?>
+                                </a>
+                            </li>
+                        <?php endforeach; ?>
+                    </ul>
+                <?php else: ?>
+                    <p><?= $this->translate('This course does not have any subcourses.') ?></p>
+                <?php endif; ?>
+            </div>
+            <div class="col-md-8">
+                <h2><?= $this->translate('Exams and summaries') ?></h2>
+                <ul>
+                    <?php
+                    // Doctrine `Collections` are `Traversable`s, for which we can use `iterator_to_array` instead of
+                    // `array_push` or `foreach`. Appending also faster than unshifting.
+                    $courses = iterator_to_array($course->getChildren());
```

Comment:
I propose changing this to the following to keep the original order while still putting the main course up front:
```
                    $courses = iterator_to_array($course->getChildren());
                    $coursesReverse = array_reverse($courses);
                    $coursesReverse[] = $course;
                    $courses = array_reverse($coursesReverse);
```

Reply:
Is there really a need to keep the original ordering? As there isn't a strict order at this point, it is just the order in which the exams/summaries are added to the website. If someone decides to upload old exams, the order on the website as it currently is will also be wrong.

Reply:
I think that front-end should adhere to the ordering provided by back-end.

Reply:
In that case I'll change it to `array_unshift`, that should be slightly faster than reversing the array twice.